pandas/io/pytables.py

```diff
@@ -590,16 +590,35 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_value, traceback):
         self.close()
 
-    def keys(self) -> List[str]:
+    def keys(self, kind: Optional[str] = "pandas") -> List[str]:
```

Comment:
Hi,
Thanks very much for putting it together!
Why not make "tables" the default and worry about the performance later? (Some "tables"/"table" mixup in the docs btw.)
`dask` for example would have to use the keyword to work properly, which would make it incompatible with
any `pandas < 1.0.x` that doesn't have this keyword.
Given that, I am not so sure anymore to solve that here and in this way, it is an API change that will most certainly break "userspace", but I'd leave that to the maintainers.

Reply:
Making "tables" the default is probably breaking the API, but I can do some experiments by changing it and running the unit tests and see what breaks :smile:

Reply:
Another suggestion that was previously mentioned is to fall back to the "tables" option if the "pandas" option returns no results. This assumes that we either use an HDF5 file that was produced by pandas and only has pandas style tables, or that we read in an HDF5 file that was produced by another application and has HDF5 style tables.
I think that in this way we stay backwards compatible *and* we do not have the performance issue with the pandas style tables.

Reply:
Hi,
Again, thanks for taking it on and coming up with a solution.
Looks good to me, so we'll see.
Chhers.

Reply:
I've just come across this as it's stopping me from working with HDF5 files not produced by pandas.

My issue - how will this cope with files that were produced in some upstream system conforming to the HDF5 specification, then data added to it with pandas (so you have a mix)

Surely the real fix is for pandas to conform to the HDF5 specification (so files work with h5py)

Reply:
pandas has a specific format of meta-data on top of HDF5. This is not likely to change.